ThreadGroup/Controllers/TGRouterAuthViewController.m

```diff
@@ -28,6 +29,7 @@ - (void)viewWillAppear:(BOOL)animated {
 
 - (void)resetView {
     self.errorLabel.hidden = YES;
```

Comment:
An additional thing you could do is highlight the button when it is disabled. Clearly shows to the user that the button is disabled.